### PR TITLE
Add /_info route to realm server

### PR DESCRIPTION
**Pull request by lukemelia**
- Returns JSON-API doc with realm metadata (name for now) that is configured via a .realm.json file at the root of the realm

### CODE REVIEW

File: packages/runtime-common/realm.ts [hunk 1]
```diff
@@ -804,6 +805,36 @@ export class Realm {
     });
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  private async realmInfo(_request: Request): Promise<Response> {
```
Comment (backspace) [inline packages/runtime-common/realm.ts:808]: Should we add [this](https://github.com/cardstack/boxel/blob/dfc8df68cd6deed7a9e9111ab4c64f0eff6025c9/packages/host/.eslintrc.js#L27) for this package?

File: packages/realm-server/tests/realm-server-test.ts [hunk 2]
```diff
@@ -518,6 +540,14 @@ module('Realm Server serving from root', function (hooks) {
           id: testRealmHref,
           type: 'directory',
           relationships: {
+            '.realm.json': {
```
Comment (backspace) [inline packages/realm-server/tests/realm-server-test.ts]: This [isn’t showing](https://github.com/cardstack/boxel/actions/runs/4932502848/jobs/8815593336?pr=329#step:8:158) as expected 😯

File: packages/host/tests/integration/search-index-test.ts [hunk 3]
```diff
@@ -2487,6 +2487,21 @@ posts/please-ignore-me.json
...
+        '.realm.json': `{ name: 'Example Workspace' }`,
+        'post.json': { data: { meta: { adoptsFrom: baseCardRef } } },
+      },
+      this.owner
+    );
+
+    await realm.ready;
+    let indexer = realm.searchIndex;
+    let card = await indexer.card(new URL(`${testRealmURL}post`));
+    assert.ok(card, 'instance exists');
```
Comment (habdelra) [inline packages/host/tests/integration/search-index-test.ts:2502]: I'm unsure how this test proves that the search index ignored the .realm.json file. shouldn't you try to load the `.realm.json` file from the search index and assert that it is not there?

File: packages/runtime-common/realm.ts [hunk 4]
```diff
@@ -804,6 +805,36 @@ export class Realm {
     });
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (habdelra) [inline packages/runtime-common/realm.ts]: I'm unclear why you needed to add this eslint disable, our lint rules should be setup such that params prefixed with `_` are allowed to be unsued.

File: packages/runtime-common/realm.ts [hunk 5]
```diff
@@ -804,6 +805,36 @@ export class Realm {
...
+      data: {
+        id: this.paths.url.toString(),
+        type: 'realm-info',
+        attributes: {
+          name,
+        },
+      },
+    };
+    return createResponse(JSON.stringify(doc, null, 2), {
+      headers: { 'content-type': SupportedMimeType.CardJson },
```
Comment (habdelra) [inline packages/runtime-common/realm.ts]: this is not card json--card json has the `meta.adoptsFrom` (notably it should pass this assertion test https://github.com/cardstack/boxel/blob/main/packages/runtime-common/card-document.ts#L187-L189). rather, this is just normal application/vnd.api+json

File: packages/realm-server/tests/realm-server-test.ts (hunk 2 above)
Comment (lukemelia) [reply packages/realm-server/tests/realm-server-test.ts]: Removed it.

File: packages/host/tests/integration/search-index-test.ts (hunk 3 above)
Comment (lukemelia) [reply packages/host/tests/integration/search-index-test.ts:2502]: Updated.

File: packages/runtime-common/realm.ts (hunk 4 above)
Comment (lukemelia) [reply packages/runtime-common/realm.ts]: Fixed this eslint config

**Review by habdelra — APPROVED** [review]

**Review by backspace — APPROVED** [review]

**Review by burieberry — APPROVED** [review]

**Review by jurgenwerk — APPROVED** [review]

**Pull request merged**